sortinghat/core/jobs.py

```diff
@@ -213,6 +213,126 @@ def affiliate(ctx, uuids=None):
+    :param source_uuids: list of individuals identifiers to look matches for
+    :param target_uuids: list of individuals identifiers where to look for matches
+    :param criteria: list of fields which the unify will be based on
+        (`email`, `name` and/or `username`)
+
+    :returns: a list with which individuals were merged
+        and the errors found running the job
+    """
+    def _group_recommendations(recs):
+        """Calculate unique sets of identities from matching recommendations.
+
+        For instance, given a list of matching groups like
+        A = {A, B}; B = {B,A,C}, C = {C,} and D = {D,} the output
+        for keys A, B and C will be the group {A, B, C}. As D has no matches,
+        it won't be included in any group and it won't be returned.
+
+        :param recs: recommendations of matching identities
+
+        :returns: a list including unique groups of matches
+        """
+        groups = []
+        for group_key in recs:
+            g_uuids = recs[group_key]
+            if group_key not in g_uuids:
+                g_uuids.append(group_key)
+            g_uuids = sorted(g_uuids)
+            if (len(g_uuids) > 1) and (g_uuids not in groups):
+                groups.append(g_uuids)
+        return groups
+
+    if not target_uuids:
+        target_uuids = Individual.objects.values_list('mk', flat=True).iterator()
+
+    results = []
+    errors = []
+
+    job_result = {
+        'results': results,
+        'errors': errors
+    }
+
+    engine = RecommendationEngine()
+
+    # Create a new context to include the reference
+    # to the job id that will perform the transaction.
+    job = rq.get_current_job()
+    job_ctx = SortingHatContext(ctx.user, job.id)
+
+    trxl = TransactionsLog.open('unify', job_ctx)
+
+    match_recs = {}
+    for rec in engine.recommend('matches', source_uuids, target_uuids, criteria):
+        match_recs[rec.key] = list(rec.options)
+
+    match_groups = _group_recommendations(match_recs)
+
+    # Apply the merge of the matching identities
+    for group in match_groups:
+        uuid = group[0]
+        result = group[1:]
+        merged_to, errs = _merge_individuals(job_ctx, uuid, result)
+        if merged_to:
+            results.append(merged_to)
+        errors.extend(errs)
+
+    trxl.close()
+
+    return job_result
+
+
+def _merge_individuals(job_ctx, source_indv, target_indvs):
+    """Merge a set of individuals.
+
+    Returns a tuple with two elements: list of the uuids from
+    the individuals who were merged; list of errors found
+    during the process.
+
+    :param job_ctx: job context
+    :param source_indv: valid individual identifier where
+        the rest of individuals will be merged to
+    :param target_indvs: list of identifiers of the individuals
+        who will be merged with the source individual
+
+    :returns: tuple with the uuids from the individuals who
+        were merged and list of errors found during the process
+    """
```

Comment:
This comment doesn't match with what the function is returning. It only returns one uuid.